lib/widgets/new_dm_sheet.dart

```diff
@@ -0,0 +1,331 @@
+        reverse: true,
+        child: Row(children: [
+          Expanded(child: Wrap(
+            spacing: 6,
+            runSpacing: 4,
+            crossAxisAlignment: WrapCrossAlignment.center,
+            children: [
+              ...selectedUserIds.map((userId) => _SelectedUserChip(userId: userId)),
+              IntrinsicWidth(child: _buildSearchField(context)),
+            ])),
+        ])));
+  }
+
+  Widget _buildSearchField(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+    final zulipLocalizations = ZulipLocalizations.of(context);
+    final hintText = selectedUserIds.isEmpty
+      ? zulipLocalizations.newDmSheetSearchHintEmpty
+      : zulipLocalizations.newDmSheetSearchHintSomeSelected;
+
+    return TextField(
+      controller: searchController,
+      cursorColor: designVariables.foreground,
+      style: TextStyle(color: designVariables.textMessage,
+        fontSize: 17, height: 22 / 17),
+      scrollPadding: EdgeInsets.zero,
+      decoration: InputDecoration(
+        isDense: true,
+        contentPadding: EdgeInsets.zero,
+        hintText: hintText,
+        hintStyle: TextStyle(color: designVariables.labelSearchPrompt,
+          fontSize: 17, height: 22 / 17),
+        border: InputBorder.none));
+  }
+}
+
+class _SelectedUserChip extends StatelessWidget {
+  const _SelectedUserChip({
+    required this.userId
+  });
+
+  final int userId;
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+    final store = PerAccountStoreWidget.of(context);
+    final user = store.getUser(userId)!;
+
+    return Container(
+    decoration: BoxDecoration(color: designVariables.bgMenuButtonSelected, borderRadius: BorderRadius.circular(3)),
+    child: Row(
+      mainAxisSize: MainAxisSize.min,
+      children: [
+        Avatar(userId: userId, size: 22, borderRadius: 3),
+        Padding(
+          padding: const EdgeInsetsDirectional.fromSTEB(5, 3, 4, 3),
+          child: Text(user.fullName,
+            style: TextStyle(
+              fontSize: 16,
+              height: 16 / 16,
+              color: designVariables.labelMenuButton))),
+      ]));
+  }
+}
+
+class _NewDmUserList extends StatelessWidget {
+  const _NewDmUserList({
+    required this.filteredUsers,
+    required this.selectedUserIds,
+    required this.onUserTapped
+  });
+
+  final List<User> filteredUsers;
+  final Set<int> selectedUserIds;
+  final void Function(int userId) onUserTapped;
+
+  @override
+  Widget build(BuildContext context) {
+    final designVariables = DesignVariables.of(context);
+
+    return ListView.builder(
+      padding: EdgeInsets.all(8),
+      itemCount: filteredUsers.length,
+      itemBuilder: (context, index) {
+        final user = filteredUsers[index];
+        final isSelected = selectedUserIds.contains(user.userId);
+
+        return InkWell(
+          onTap: () => onUserTapped(user.userId),
```

Comment:
Let's use `splashFactory: NoSplash.splashFactory,` here; in general we avoid having Material's splash effect in the app.